src/commands/services/plan.ts

```diff
@@ -8,21 +9,96 @@ export default class Plan extends Command {
   static usage = 'services:plan [ID] [OPTIONS]';
 
   static args = [
-    { name: 'id', description: 'Service Id', required: true }
+    { name: 'service_id', description: 'Service Id' },
+    { name: 'service_version', description: 'Service Version' },
+    { name: 'environment_id', description: 'Environment Id' }
   ];
 
   static flags = {
-    help: flags.help({ char: 'h' }),
-    file: flags.string({ char: 'f' })
+    help: flags.help({ char: 'h' })
   };
 
   async run() {
-    const { args, flags } = this.parse(Plan);
-    const { data: template } = await this.architect.get(`/registry/repositories/${args.id}/plan`);
-    if (flags.file) {
-      await fs.writeJson(flags.file, template, { spaces: 2 });
+    const answers = await this.promptOptions();
+
+    let plan: any;
+    const tasks = new Listr([
+      {
+        title: `Planning`,
+        task: async () => {
+          const params = {
+            environment_id: answers.environment_id,
+            service_version: answers.service_version,
+          };
+          const { data } = await this.architect.get(`/repositories/${answers.service_id}/plan`, { params });
+          plan = data;
+        }
+      }
+    ]);
+    await tasks.run();
+    this.log(plan.info);
+
+    const confirmation = await inquirer.prompt({
+      type: 'confirm',
+      name: 'apply',
+      message: 'Would you like to apply this plan?'
+    } as inquirer.Question);
+
+    if (confirmation.apply) {
+      const tasks = new Listr([
+        {
+          title: `Applying`,
+          task: async () => {
+            const params = {
+              environment_id: answers.environment_id,
+              timestamp: plan.timestamp
+            };
+            await this.architect.get(`/repositories/${answers.service_id}/apply`, { params });
+          }
+        }
+      ]);
+      await tasks.run();
     } else {
-      this.styled_json(template);
+      this.warn('Canceled apply');
     }
   }
+
+  async promptOptions() {
+    const { args, flags } = this.parse(Plan);
+
+    inquirer.registerPrompt('autocomplete', require('inquirer-autocomplete-prompt'));
+
+    const answers = await inquirer.prompt([{
```

Comment:
These questions should probably belong to a inquirer utils since they will be repeated in a lot fo these cmds.